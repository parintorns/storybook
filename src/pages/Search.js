import React, { Component } from 'react'
import PropTypes from 'prop-types'

import {
  Container,
  Header,
  Title,
  Left,
  Right,
  Body
} from 'native-base'

export class Search extends Component {
  render () {
    return (
      <Container>
        <Header>
          <Left />
          <Body>
            <Title>Search</Title>
          </Body>
          <Right />
        </Header>
      </Container>
    )
  }
}

export default Search
