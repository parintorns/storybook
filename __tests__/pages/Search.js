import React from 'react'
import { shallow } from 'enzyme'

import Search from '../../src/pages/Search'

describe('Render', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <Search />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
