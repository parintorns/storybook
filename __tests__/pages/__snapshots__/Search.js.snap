// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Header)
        virtual={undefined}
      >
        <Styled(Left)
          virtual={undefined}
        />
        <Styled(Body)
          virtual={undefined}
        >
          <Styled(Title)
            virtual={undefined}
          >
            Search
          </Styled(Title)>
        </Styled(Body)>
        <Styled(Right)
          virtual={undefined}
        />
      </Styled(Header)>,
      "virtual": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(Left)
            virtual={undefined}
          />,
          <Styled(Body)
            virtual={undefined}
          >
            <Styled(Title)
              virtual={undefined}
            >
              Search
            </Styled(Title)>
          </Styled(Body)>,
          <Styled(Right)
            virtual={undefined}
          />,
        ],
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Styled(Title)
              virtual={undefined}
            >
              Search
            </Styled(Title)>,
            "virtual": undefined,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Search",
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Search",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "virtual": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Header)
          virtual={undefined}
        >
          <Styled(Left)
            virtual={undefined}
          />
          <Styled(Body)
            virtual={undefined}
          >
            <Styled(Title)
              virtual={undefined}
            >
              Search
            </Styled(Title)>
          </Styled(Body)>
          <Styled(Right)
            virtual={undefined}
          />
        </Styled(Header)>,
        "virtual": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Styled(Left)
              virtual={undefined}
            />,
            <Styled(Body)
              virtual={undefined}
            >
              <Styled(Title)
                virtual={undefined}
              >
                Search
              </Styled(Title)>
            </Styled(Body)>,
            <Styled(Right)
              virtual={undefined}
            />,
          ],
          "virtual": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Styled(Title)
                virtual={undefined}
              >
                Search
              </Styled(Title)>,
              "virtual": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Search",
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Search",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "virtual": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
